### ISSUE TITLE
Generic `SimpleObject`, `concrete` and lifetime.

### ISSUE THREAD

**Issue by BratSinot** [async-graphql/async-graphql#774]
Greetings!

Faced with problem of using generic as `SimpleObject`.

## Actual Behavior
```log
11 | #[derive(SimpleObject)]
   |          ^^^^^^^^^^^^ help: indicate the anonymous lifetime: `S'_, impleObject`
```

## Steps to Reproduce the Problem
```rust
#[derive(SimpleObject)]
#[graphql(concrete(name = "Bar0", params(Bar0)))]
#[graphql(concrete(name = "Bar1", params(Bar1)))]
struct Foo<'a, T>
where
    T: Sync + OutputType + 'a,
{
    data: &'a T,
}
```

**Comment by sunli829** [comment]
Fixed in `3.0.20`

**Comment by BratSinot** [comment]
> Fixed in `3.0.20`

Thanks a lot!
